src/Kentico.Xperience.Lucene/Services/Implementations/DefaultLuceneTaskProcessor.cs

```diff
@@ -1,58 +1,93 @@
+using CMS.Base;
 using CMS.Core;
+using CMS.Websites;
 using Kentico.Xperience.Lucene.Models;
+using Lucene.Net.Documents;
+using Lucene.Net.Documents.Extensions;
 
 namespace Kentico.Xperience.Lucene.Services;
 
+internal class LuceneBatchResult
+{
+    internal int SuccessfulOperations { get; set; } = 0;
+    internal HashSet<LuceneIndex> PublishedIndices { get; set; } = new();
+}
+
 internal class DefaultLuceneTaskProcessor : ILuceneTaskProcessor
 {
+    internal const string URL_NAME = "Url";
```

Comment:
While the `DefaultLuceneTaskProcessor` should be `internal`, I think any field names of fields that we automatically add to the document should be `public`.

Reply:
Added class for this purpose - BaseProperties